modules/aioseop_sitemap.php

```diff
@@ -3149,6 +3178,57 @@ public function get_addl_pages() {
 			}
 			$pages = apply_filters( $this->prefix . 'addl_pages', $pages );
 
+			$pages = $this->remove_addl_static_pages( $pages );
+
+			return $pages;
+		}
+
+		/**
+		 * The remove_addl_static_pages() function.
+		 *
+		 * Removes the homepage/posts page from the Additional Pages index if it is static - #2126.
+		 *
+		 * @since 3.2.0
+		 *
+		 * @param array $pages
+		 * @return array $pages
+		 */
+		private function remove_addl_static_pages( $pages ) {
+			$pages_to_remove = array();
+			if ( 0 !== get_option( 'page_on_front' ) ) {
+				$homepage_url = get_site_url() . '/';
+				array_push( $pages_to_remove, $homepage_url );
+			}
+
+			$static_posts_page_id = get_option( 'page_for_posts' );
+			if ( 0 !== $static_posts_page_id ) {
+				array_push( $pages_to_remove, get_permalink( $static_posts_page_id ) );
+			}
+
+			if ( count( $pages_to_remove ) > 0 ) {
+				return $this->remove_urls_from_sitemap_page( $pages, $pages_to_remove );
+			}
+			return $pages;
+		}
+
+		/**
+		 * The remove_urls_from_sitemap_page() function.
+		 *
+		 * Removes URLs from a sitemap page. This is used both for indexes and pages within indexes.
+		 *
+		 * @since 3.2.0
+		 *
+		 * @param array $pages
+		 * @param array $pages_to_remove
+		 * @return array $pages
+		 */
+		private function remove_urls_from_sitemap_page( $pages, $pages_to_remove ) {
+			$count = count( $pages );
```

Comment:
This operation could potentially improve processing by changing it to. 

```
foreach ( $pages_to_remove as $page_url_to_remove ) {
	$key = array_search( $page_url_to_remove, array_column( $pages, 'loc' ) );
	if ( $key ) {
		unset( $pages[ $key ] );
	}
}
```

OR

```
foreach ( $pages_to_remove as $page_url_to_remove ) {
	if ( in_array( $page_url_to_remove, array_column( $pages, 'loc' ) ) ) {
		unset( $pages[ array_search( $page_url_to_remove, array_column( $pages, 'loc' ) ) ] );
	}
}
```

It would make debugging a little quicker though with the fewer loops.

Reply:
We'll leave this as is as discussed since I can't get this to work.